### DISCUSSION TITLE
Doing a nested create. Github CoPilot helped me, but I need help understanding it.

### DISCUSSION THREAD

**Discussion by gmcamposano** [prisma/prisma#17027] [Q&A]
**Numero_de_factura** is basically the invoice info. 
**Comprobante_fiscal && Consumidor_final** are the invoice numbers types

```
model Numero_de_factura {
  id                 String               @id @default(uuid())
  rnc_empresa        Int?
  nombre_empresa     String?
  factura            Factura[]
  comprobante_fiscal Comprobante_fiscal[]
  consumidor_final   Consumidor_final[]
}

model Factura {
  id                   String            @id @default(uuid())
  cantidad             Int
  total                Int
  numero_de_factura_id String
  numero_de_factura    Numero_de_factura @relation(fields: [numero_de_factura_id], references: [id])
}

model Comprobante_fiscal {
  id                   Int                @id @default(autoincrement())
  numero_de_factura_id String?
  numero_de_factura    Numero_de_factura? @relation(fields: [numero_de_factura_id], references: [id])
}

model Consumidor_final {
  id                   Int                @id @default(autoincrement())
  numero_de_factura_id String?
  numero_de_factura    Numero_de_factura? @relation(fields: [numero_de_factura_id], references: [id])
}
```

**What am i trying to do?** If `req.body` returns in `datos.tipo_de_factura` a string called `consumidor_final` or `comprobante_fiscal`, i would like to increment the number of that given table. I am using a dynamic key to set the table name, but after that point i have no idea what is happening. I don't know how to access that table to increment the id because i dont have access to the `numero_de_factura_id` that is used there. Any idea what github copilot suggested there and why it works?
```
const { datos } = req.body

    let tipo =
      datos.tipo_de_factura === 'consumidor-final'
        ? 'consumidor_final'
        : 'comprobante_fiscal'

    const orden = await prisma.numero_de_factura.create({
      data: {
        ...req.body.datos,
        factura: {
          createMany: {
            data: req.body.factura
          }
        },
        [tipo]: {
          create: {
            ...req.body[tipo]
          }
        }
      }
    })
```

**Comment by Howardzhong2021** [comment]
这是来自QQ邮箱的自动回复邮件。您好，您的回复我已经收到，感谢您的关心与体谅，我将尽快回复您。

**Comment by gmcamposano** [comment]
Can somebody help me with this one?

**Reply by nurul3101** [reply]
Hello @gmcamposano 👋 
Are you trying to increment the value of `id` column in the `Comprobante_fiscal` and `Consumidor_final` models based on the value that you get in the request body?

**Reply by gmcamposano** [reply]
Yes! The table's only purpose is to increase +1 depending on the invoice type. If comprobante_fiscal was selected, then add 1, if the other, then add 1 to that one. That way I can keep track of that number.
If you have another solution which you think works better, please feel free to suggest it.

**Reply by gmcamposano** [reply]
I think I found out after a quite some time:
```
[tipo]: {
          create: { } //this would basically do the same thing, which means that ...req.body[tipo] is undefined
        }
```
I am beginning to think this is not a good approach of what I am trying to solve. @nurul3101 how would you do it?

**Reply by nurul3101** [reply]
If I understand correctly, you are changing the value of a primary key `id` by incrementing it. Ideally, you should not change the value of id and rather have another field which is just used for maintaining the count, and you can use [increment](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#increment-all-view-and-likes-fields-of-all-post-records-by-1) option to increment the count.

**Reply by gmcamposano** [reply]
I dont want to change the primary key directly, I am using autoincrement for it. I just wanted to find out how can I increment it if I did not have anything else than the id in that table, but as I told you on my previous comment, if i create an empty object it seems to do the trick.

**Reply by nurul3101** [reply]
Passing an empty object to create clause shouldn't cause any issues in my opinion.

**Comment by Howardzhong2021** [comment]
这是来自QQ邮箱的自动回复邮件。您好，您的回复我已经收到，感谢您的关心与体谅，我将尽快回复您。

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
